proposals/NNNN-non-escapable.md

```diff
@@ -0,0 +1,442 @@
+```swift
+extension Wrapper: Escapable where T: ~Copyable {}
+```
+is exactly the same as
+```swift
+extension Wrapper: Escapable where T: Escapable & ~Copyable {}
+```
+which means that `Wrapper` is `Escapable` whenever `T` is `Escapable` regardless of whether `T` is `Copyable` or not.
+
+## Source compatibility
+
+The compiler will treat any type without explicit `~Escapable` as escapable.
+This matches the current behavior of the language.
+
+Only when new types are marked as `~Escapable` does this have any impact.
+
+Adding `~Escapable` to an existing concrete type is generally source-breaking because existing source code may rely on being able to escape values of this type.
+Removing `~Escapable` from an existing concrete type is not generally source-breaking since it effectively adds a new capability, similar to adding a new protocol conformance.
+
+## ABI compatibility
+
+As above, existing code is unaffected by this change.
+Adding or removing a `~Escapable` constraint on an existing type is an ABI-breaking change.
+
+## Implications on adoption
+
+Manglings and interface files will only record the lack of escapability.
+This means that existing interfaces consumed by a newer compiler will treat all types as escapable.
+Similarly, an old compiler reading a new interface will have no problems as long as the new interface does not contain any `~Escapable` types.
+
+These same considerations ensure that escapable types can be shared between previously-compiled code and newly-compiled code.
+
+Retrofitting existing generic types so they can support both escapable and nonescapable type arguments is possible with care.
+
+## Future directions
+
+#### `StorageView` type
+
+This proposal is being driven in large part by the needs of the `StorageView` type that has been discussed elsewhere.
+Briefly, this type would provide an efficient universal “view” of array-like data stored in contiguous memory.
+Since values of this type do not own any data but only refer to data stored elsewhere, their lifetime must be limited to not exceed that of the owning storage.
+We expect to publish a sample implementation and proposal for that type very soon.
+
+#### Lifetime dependency annotations
+
+Nonescapable types have a set of inherent restrictions on how they can be passed as arguments, stored in variables, or returned from functions.
+A companion proposal builds on this by supporting more detailed annotations that link the lifetimes of different objects.
+This would allow, for example, a container to vend an iterator value that held a direct unmanaged pointer to the container's contents.
+The lifetime dependency would ensure that such an iterator could not outlive the container to whose contents it referred.
+
+```swift
+// Example: Nonescaping iterator
+struct NEIterator {
+  // `borrow(container)` indicates that the constructed value
+  // cannot outlive the `container` argument.
+  init(over container: MyContainer) -> borrow(container) Self {
```

Comment:
Suggest updating syntax to match companion proposal :)

Reply:
Done.  (Thanks for the feedback!)